specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingSubscription.json

```diff
@@ -400,244 +749,605 @@
+            "SubscriptionNotActive",
+            "SubscriptionHasReservations",
+            "SubscriptionTypeNotSupported",
+            "InvoiceSectionIsRestricted"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BillingSubscriptionErrorCodes",
+            "modelAsString": true
+          }
+        },
+        "message": {
+          "description": "The error message.",
+          "type": "string"
+        },
+        "details": {
+          "description": "Detailed error message explaining the error.",
+          "type": "string"
+        }
       }
     },
-    "TransferBillingSubscriptionRequestProperties": {
+    "MoveBillingSubscriptionRequest": {
       "description": "Request parameters to transfer billing subscription.",
+      "required": [
+        "destinationInvoiceSectionId"
+      ],
+      "type": "object",
       "properties": {
         "destinationInvoiceSectionId": {
           "description": "The destination invoice section id.",
+          "minLength": 1,
           "type": "string"
         }
-      },
-      "required": [
-        "destinationInvoiceSectionId"
-      ],
-      "type": "object"
+      }
     },
-    "ValidateSubscriptionTransferEligibilityError": {
-      "description": "Error details of the transfer eligibility validation",
+    "MoveEligibilityResult": {
+      "description": "Result of the transfer eligibility validation.",
+      "type": "object",
       "properties": {
-        "code": {
-          "$ref": "#/definitions/SubscriptionTransferValidationErrorCode",
-          "description": "Error code for the product transfer validation."
+        "isMoveEligible": {
+          "description": "Specifies whether the subscription is eligible to be transferred.",
+          "type": "boolean"
         },
-        "details": {
-          "description": "Detailed error message explaining the error.",
+        "errorDetails": {
+          "description": "Error details of the transfer eligibility validation.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/MoveBillingSubscriptionErrorDetails"
+            }
+          ]
+        }
+      }
+    },
+    "NextBillingCycleDetails": {
+      "description": "Billing cycle details of the product.",
+      "type": "object",
+      "properties": {
+        "billingFrequency": {
+          "description": "Billing frequency of the product under the subscription.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RenewalTermDetails": {
+      "description": "Details for the subscription's next renewal term.",
+      "type": "object",
+      "properties": {
+        "billingFrequency": {
+          "description": "The billing frequency in ISO8601 format of product in the subscription. Example: P1M, P3M, P1Y",
           "type": "string"
         },
-        "message": {
-          "description": "The error message.",
+        "productId": {
```

Comment:
(If for whatever reason we have to keep using BigIds, can we make sure we're consistent between using productId/productTypeId? Or are they different semantically?)

Reply:
@ramkd12 @amberbhargava to comment on this